packages/browser/src/provider.ts

```diff
@@ -0,0 +1,268 @@
+import { EventEmitter } from 'events';
+import type { JSONRPCRequest, JSONRPCResponseHandler, Subprovider } from '@bitski/provider-engine';
+import type { BufferBlock } from '@bitski/provider-engine/dist/modules/util/block-tracker';
+import type { BitskiEngine } from 'bitski-provider';
+import type { BitskiSDK, ProviderOptions } from './-private/sdk';
+import { ProviderError } from 'bitski-provider/lib/errors/provider-error';
+
+interface RequestArguments {
+  method: string;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  params?: unknown[] | object;
+}
+
+interface Web3Provider {
```

Comment:
Do we need a custom interface here or can we depend on something like from ethers.js?

Reply:
I'd rather not pull in a separate dependency, especially because they don't all seem to be super consistent

Reply:
It's unclear if Ethers exports interfaces or if it just exports implementations/abstract classes